user-interface/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/ExperimentDetailsComponent.java

```diff
@@ -451,47 +456,59 @@ private void onExperimentalGroupEditConfirmed(
+        experimentId);
     deletionService.deleteAllExperimentalGroups(experimentId).onError(error -> {
       throw new ApplicationException(
           "Could not edit experiments because samples are already registered.");
     });
-    addExperimentalGroups(experimentalGroupContents);
+    var result = addExperimentalGroups(experimentalGroupContents);
+    if(result.isError()) {
+      // reroll by re-adding the original groups
+      experimentInformationService.addExperimentalGroupsToExperiment(
+          experimentId, originalGroups);
+      handleGroupModificationError(experimentalGroupContents, result.getError());
+    }
   }
 
-  private void addExperimentalGroups(
+  private Result<Collection<ExperimentalGroup>, ResponseCode> addExperimentalGroups(
       Collection<ExperimentalGroupContent> experimentalGroupContents) {
     List<ExperimentalGroupDTO> experimentalGroupDTOS = experimentalGroupContents.stream()
         .map(this::toExperimentalGroupDTO).toList();
     ExperimentId experimentId = context.experimentId().orElseThrow();
     Result<Collection<ExperimentalGroup>, ResponseCode> result = experimentInformationService.addExperimentalGroupsToExperiment(
         experimentId, experimentalGroupDTOS);
-    if (result.isError()) {
-      if (result.getError().equals(ResponseCode.CONDITION_EXISTS)) {
-        throw new ApplicationException("Duplicate experimental group was selected",
-            ErrorCode.DUPLICATE_GROUP_SELECTED,
-            ErrorParameters.empty());
-      }
-      if (result.getError().equals(ResponseCode.EMPTY_VARIABLE)) {
-        throw new ApplicationException("No experimental variable was selected",
-            ErrorCode.NO_CONDITION_SELECTED,
-            ErrorParameters.empty());
-      } else {
-        throw new ApplicationException(
-            "Could not save one or more experimental groups %s %nReason: %s".formatted(
-                Arrays.toString(
-                    experimentalGroupContents.toArray()), result.getError()));
-      }
+    return result;
+  }
+
+  private void handleGroupModificationError(Collection<ExperimentalGroupContent> experimentalGroupContents,
+      ResponseCode responseCode) {
+    if (responseCode.equals(ResponseCode.CONDITION_EXISTS)) {
+      throw new ApplicationException("Duplicate experimental group was selected",
+          ErrorCode.DUPLICATE_GROUP_SELECTED,
+          ErrorParameters.empty());
+    }
+    if (responseCode.equals(ResponseCode.EMPTY_VARIABLE)) {
+      throw new ApplicationException("No experimental variable was selected",
+          ErrorCode.NO_CONDITION_SELECTED,
+          ErrorParameters.empty());
+    } else {
+      throw new ApplicationException(
+          "Could not save one or more experimental groups %s %nReason: %s".formatted(
+              Arrays.toString(
+                  experimentalGroupContents.toArray()), responseCode));
     }
```

Comment:
I'd recommend to remove this entirely, this should be handled via the message.properties not dependent on the Result object. The service can just throw the application exception with the correct errorcode